deque.go

```diff
@@ -146,21 +142,20 @@ func (q *Deque) Rotate(n int) {
 		return
 	}
 
-	modBits := len(q.buf) - 1
 	// If no empty space in buffer, only move head and tail indexes.
 	if q.head == q.tail {
 		// Calculate new head and tail using bitwise modulus.
-		q.head = (q.head + n) & modBits
-		q.tail = (q.tail + n) & modBits
+		q.head = q.next(q.head, n)
```

Comment:
Intentionally did not call `next` or `prev` in this function to avoid recalculation of `modbits`